src/DataStructures/IndexIterator.hpp

```diff
@@ -48,13 +48,13 @@ class IndexIterator {
   /// currently represents
   const Index<Dim>& operator()() const noexcept { return index_; }
 
-  /// Get the offset into a 1D array whose first dimension is assumed to vary
-  /// fastest
-  size_t offset() const noexcept { return offset_; }
+  /// Get the collapsed_index into a 1D array whose first dimension is assumed
```

Comment:
Get the collapsed index into the 1D array of the data corresponding to the current Index of the IndexIterator.  Note that the first dimension of the Index varies fastest when computing the collapsed index.